framework/jagereye_ng/io/obj_storage.py

```diff
@@ -0,0 +1,111 @@
+"""Utilities for I/O manipulation."""
+
+import json
+import os
+from io import BytesIO
+
+import boto3
+import cv2
+from botocore.client import ClientError
+
+from jagereye_ng.util.generic import get_config
+
+
+def _gen_public_read_policy(bucket_name):
+    """Generate a bucket policy to be public readable.
+
+    Args:
+      bucket_name (string): The bucket name for the policy.
+    """
+    policy = {
+        "Version": "2012-10-17",
+        "Statement": [{
+            "Sid": "",
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": ["*"]
+            },
+            "Action": ["s3:GetObject"],
+            "Resource": ["arn:aws:s3:::{}/*".format(bucket_name)]
+        }]
+    }
+
+    return json.dumps(policy)
+
+
+def save_obj(key, obj):
+    """Save an object to object store.
+
+    Args:
+      key (string): The key of the object.
+      obj (bytes|bytearray|file-like object): The object to be saved.
+    """
+    # Get configuration for object storage.
+    config = get_config()["services"]["obj_storage"]
+    endpoint_url = config["params"]["endpoint_url"]
+    bucket_name = config["params"]["bucket_name"]
+    access_key = config["credentials"]["access_key"]
+    secret_key = config["credentials"]["secret_key"]
+
+    # Connect to the object store.
+    client = boto3.client("s3",
```

Comment:
Does this mean that save operation needs to establish connection every single time? Can we save the connection for the first time it's established and re-use it again for the following save operation?